libs/backend/src/websocket/host-game.ts

```diff
@@ -0,0 +1,27 @@
+import mongoose from "mongoose";
+import { ValueOfGameEvent } from "types";
+import { joinGame } from "./join-game.js";
+import { WebSocketGamesMap } from "@/types/games.js";
+import { WebSocket } from "@fastify/websocket";
+
+export function hostGame({
+	userId,
+	socket,
+	username,
+	event,
+	games
+}: {
+	socket: WebSocket;
+	event: ValueOfGameEvent;
+	username: string;
+	userId: string;
+	games: WebSocketGamesMap;
+}) {
+	console.log("hosting a game");
+
+	const randomId = new mongoose.Types.ObjectId().toString();
+
+	games.set(randomId, new Map());
+
+	joinGame({ gameId: randomId, userId, socket, username, event, games });
```

Comment:
maybe you could combine these 2 steps, although not strictly required since i think the joingame does extra steps that would be required to do anyways, instead of using maps, maybe use sets, since for each `gameId`, only one game should exist, same for users in a room, only one user in each room with a certain name

Reply:
nvm the map and set thing, should be a map